src/controllers/items.py

```diff
@@ -230,6 +231,43 @@ async def add_items(request: Request, imdb_ids: str = None) -> MessageResponse:
     return {"message": f"Added {len(valid_ids)} item(s) to the queue"}
 
 
+@router.post(
+    "/add-manually",
+    summary="Add Media Items Manually",
+    description="Add media item manually with a magnet link or infohash",
+    operation_id="add_item_manually",
+)
+async def add_item_manually(request: Request, imdb_id: str = None, input: str = None) -> MessageResponse:
+    if not imdb_id:
+        raise HTTPException(status_code=400, detail="No IMDb ID provided")
+
+    if not imdb_id.startswith("tt"):
+        raise HTTPException(status_code=400, detail="No valid IMDb ID provided")
+    
+    type, infohash = get_type_and_infohash(input)
+
+    if not infohash:
+        raise HTTPException(status_code=400, detail="No valid input provided")
+    
+    with db.Session() as session:
+        raw_title = f"Manual Torrent for {imdb_id}"
+        torrent = Torrent(
+            raw_title=raw_title,
+            infohash=infohash,
+            data=ParsedData(raw_title=raw_title),
+        )
+        item = MediaItem(
+            {"imdb_id": imdb_id, "requested_by": "riven", "requested_at": datetime.now(), "state": States.Requested}
+        )
+        stream = Stream(torrent)
+        session.add(stream)
+        session.commit()
+        item.streams = [stream]
```

Comment:
@dreulavelle I'm not that familiar with the backend. In order for a stream to download should it be added to `streams` or to `active_stream`?

Reply:
Both, fixed in latest commit